File: csharp/unittest-practice/src/test/QuesadillaTest.cs
using Moq;
using NUnit.Framework;
using unittestpractice.main;


namespace unittestpractice.test
{

    [TestFixture]
    internal class QuesadillaTest
    {
        public Quesadilla Quesadilla;
        public Mock<IQueso> MockedQueso;
        public Mock<ITortilla> MockedTortilla;
        public Mock<ITortilla> MockedTortilla1;

        [SetUp]
        public void SetUp()
        {
            Quesadilla = new Quesadilla();
            MockedQueso = new Mock<IQueso>();
            MockedTortilla = new Mock<ITortilla>();
            MockedTortilla1 = new Mock<ITortilla>();

            Quesadilla.SetQueso(MockedQueso.Object);
            Quesadilla.SetTortilla(MockedTortilla.Object);
            Quesadilla.SetTortilla1(MockedTortilla1.Object);
        }

        [TestCase]
        public void QuesadillaPerfecta()
        {
            MockedQueso.Setup(queso => queso.IsMelted()).Returns(true);
            MockedTortilla.Setup(tortilla => tortilla.IsToasted()).Returns(true);
            MockedTortilla.Setup(tortilla => tortilla.GetCurrentTemperature()).Returns(10);
            MockedTortilla.Setup(tortilla => tortilla.GetToastTemperature()).Returns(10);
            MockedQueso.Setup(queso => queso.GetCurrentTemperature()).Returns(15);
            MockedQueso.Setup(queso => queso.GetMeltingTemperature()).Returns(10);
            Assert.AreEqual("Perfect quesadilla", Quesadilla.PrepareSingle());
        }
        
        [TestCase] 
        public void QuesadillaBuena()
        {
            MockedQueso.Setup(queso => queso.IsMelted()).Returns(true);
            MockedTortilla.Setup(tortilla => tortilla.IsToasted()).Returns(false);
            MockedTortilla.Setup(tortilla => tortilla.GetCurrentTemperature()).Returns(15);
            MockedTortilla.Setup(tortilla => tortilla.GetToastTemperature()).Returns(10);
            MockedQueso.Setup(queso => queso.GetCurrentTemperature()).Returns(15);
            MockedQueso.Setup(queso => queso.GetMeltingTemperature()).Returns(10);
            Assert.AreEqual("Good quesadilla", Quesadilla.PrepareSingle());
        }

        [TestCase]
        public void QuesadillaTerrible()
        {
            MockedQueso.Setup(queso => queso.IsMelted()).Returns(false);
            MockedTortilla.Setup(tortilla => tortilla.IsToasted()).Returns(true);
            MockedTortilla.Setup(tortilla => tortilla.GetCurrentTemperature()).Returns(15);
            MockedTortilla.Setup(tortilla => tortilla.GetToastTemperature()).Returns(10);
            MockedQueso.Setup(queso => queso.GetCurrentTemperature()).Returns(8);
            MockedQueso.Setup(queso => queso.GetMeltingTemperature()).Returns(10);
            Assert.AreEqual("Terrible quesadilla", Quesadilla.PrepareSingle());
        }

        [TestCase]
        public void NoHayGas()
        {
            MockedQueso.Setup(queso => queso.IsMelted()).Returns(false);
            MockedTortilla.Setup(tortilla => tortilla.IsToasted()).Returns(false);
            MockedTortilla.Setup(tortilla => tortilla.GetCurrentTemperature()).Returns(10);
            MockedTortilla.Setup(tortilla => tortilla.GetToastTemperature()).Returns(20);
            MockedQueso.Setup(queso => queso.GetCurrentTemperature()).Returns(15);
            MockedQueso.Setup(queso => queso.GetMeltingTemperature()).Returns(10);
            Assert.AreEqual("You ran out of gas", Quesadilla.PrepareSingle());
        }

        [TestCase]
        public void QuesadillaDoblePerfecta()
        {
            MockedQueso.Setup(queso => queso.IsMelted()).Returns(true);
            MockedTortilla.Setup(tortilla => tortilla.IsToasted()).Returns(true);
            MockedTortilla.Setup(tortilla => tortilla.GetCurrentTemperature()).Returns(10);
            MockedTortilla.Setup(tortilla => tortilla.GetToastTemperature()).Returns(10);
            MockedTortilla1.Setup(tortilla1 => tortilla1.IsToasted()).Returns(true);
            MockedTortilla1.Setup(tortilla1 => tortilla1.GetCurrentTemperature()).Returns(15);
            MockedTortilla1.Setup(tortilla1 => tortilla1.GetToastTemperature()).Returns(10);
            MockedQueso.Setup(queso => queso.GetCurrentTemperature()).Returns(15);
            MockedQueso.Setup(queso => queso.GetMeltingTemperature()).Returns(10);
            Assert.AreEqual("Perfect quesadilla", Quesadilla.PrepareDouble());
        }
        [TestCase]
        public void QuesadillaDobleBuena()
        {
            MockedQueso.Setup(queso => queso.IsMelted()).Returns(true);
            MockedTortilla.Setup(tortilla => tortilla.IsToasted()).Returns(true);
            MockedTortilla.Setup(tortilla => tortilla.GetCurrentTemperature()).Returns(15);
            MockedTortilla.Setup(tortilla => tortilla.GetToastTemperature()).Returns(10);
            MockedTortilla1.Setup(tortilla1 => tortilla1.IsToasted()).Returns(false);
            MockedTortilla1.Setup(tortilla1 => tortilla1.GetCurrentTemperature()).Returns(5);
            MockedTortilla1.Setup(tortilla1 => tortilla1.GetToastTemperature()).Returns(10);
            MockedQueso.Setup(queso => queso.GetCurrentTemperature()).Returns(15);
            MockedQueso.Setup(queso => queso.GetMeltingTemperature()).Returns(10);
            Assert.AreEqual("Good quesadilla", Quesadilla.PrepareDouble());
        }

        [TestCase]
        public void QuesadillaDobleMala()
        {
            MockedQueso.Setup(queso => queso.IsMelted()).Returns(false);
            MockedTortilla.Setup(tortilla => tortilla.IsToasted()).Returns(true);
            MockedTortilla.Setup(tortilla => tortilla.GetCurrentTemperature()).Returns(15);
            MockedTortilla.Setup(tortilla => tortilla.GetToastTemperature()).Returns(10);
            MockedTortilla1.Setup(tortilla1 => tortilla1.IsToasted()).Returns(true);
            MockedTortilla1.Setup(tortilla1 => tortilla1.GetCurrentTemperature()).Returns(15);
            MockedTortilla1.Setup(tortilla1 => tortilla1.GetToastTemperature()).Returns(10);
            MockedQueso.Setup(queso => queso.GetCurrentTemperature()).Returns(5);
            MockedQueso.Setup(queso => queso.GetMeltingTemperature()).Returns(10);
            Assert.AreEqual("Bad quesadilla", Quesadilla.PrepareDouble());
        }

        [TestCase]
        public void QuesadillaDobleTerrible()
        {
            MockedQueso.Setup(queso => queso.IsMelted()).Returns(false);
            MockedTortilla.Setup(tortilla => tortilla.IsToasted()).Returns(false);
            MockedTortilla.Setup(tortilla => tortilla.GetCurrentTemperature()).Returns(8);
            MockedTortilla.Setup(tortilla => tortilla.GetToastTemperature()).Returns(10);
            MockedTortilla1.Setup(tortilla1 => tortilla1.IsToasted()).Returns(true);
            MockedTortilla1.Setup(tortilla1 => tortilla1.GetCurrentTemperature()).Returns(20);
            MockedTortilla1.Setup(tortilla1 => tortilla1.GetToastTemperature()).Returns(10);
            MockedQueso.Setup(queso => queso.GetCurrentTemperature()).Returns(5);
            MockedQueso.Setup(queso => queso.GetMeltingTemperature()).Returns(10);
            Assert.AreEqual("Terrible quesadilla", Quesadilla.PrepareDouble());
        }

        [TestCase]
        public void QuesadillaDobleOutOfGas()
        {
            MockedQueso.Setup(queso => queso.IsMelted()).Returns(false);
            MockedTortilla.Setup(tortilla => tortilla.IsToasted()).Returns(false);
            MockedTortilla.Setup(tortilla => tortilla.GetCurrentTemperature()).Returns(15);
            MockedTortilla.Setup(tortilla => tortilla.GetToastTemperature()).Returns(10);
            MockedTortilla1.Setup(tortilla1 => tortilla1.IsToasted()).Returns(false);
            MockedTortilla1.Setup(tortilla1 => tortilla1.GetCurrentTemperature()).Returns(12);
            MockedTortilla1.Setup(tortilla1 => tortilla1.GetToastTemperature()).Returns(10);
            MockedQueso.Setup(queso => queso.GetCurrentTemperature()).Returns(2);
            MockedQueso.Setup(queso => queso.GetMeltingTemperature()).Returns(10);
            Assert.AreEqual("You ran out of gas", Quesadilla.PrepareDouble());
        }
    }
}

File: python/src/test/QuesadillaTest.py
import unittest
from Quesadilla import *
from Queso import *
from Tortilla import *
from mockito import mock, verify, when


class QuesadillaTest(unittest.TestCase):
    def setUp(self):
        self.quesadilla = Quesadilla()
        self.quesadillaDoble = Quesadilla()
        self.mockedQueso = mock(Queso, strict=False)
        self.mockedQuesoDoble = mock(Queso, strict=False)
        self.mockedTortilla = mock(Tortilla, strict=False)
        self.mockedTortilla1 = mock(Tortilla, strict=False)
        self.mockedTortilla2 = mock(Tortilla, strict=False)
        self.quesadilla.setQueso(self.mockedQueso)
        self.quesadilla.setTortilla(self.mockedTortilla)
        self.quesadillaDoble.setQueso(self.mockedQuesoDoble)
        self.quesadillaDoble.setTortillas(self.mockedTortilla1,
                                          self.mockedTortilla2)

    def test_quesadillaPerfecta(self):
        when(self.mockedQueso).isMelted().thenReturn(True)
        when(self.mockedTortilla).isToasted().thenReturn(True)
        when(self.mockedTortilla).getCurrentTemperature().thenReturn(
            2, 8, 8, 8, 14)
        when(self.mockedTortilla).getToastTemperature().thenReturn(10)
        when(self.mockedQueso).getCurrentTemperature().thenReturn(
            2, 8, 8, 8, 14)
        when(self.mockedQueso).getMeltingTemperature().thenReturn(10)
        self.assertEquals("Perfect quesadilla",
                          self.quesadilla.prepareSingle())
        verify(self.mockedTortilla, times=1).toast(True)
        verify(self.mockedQueso, times=1).melt(True)

    def test_quesadillaBuena(self):
        when(self.mockedQueso).isMelted().thenReturn(True)
        when(self.mockedTortilla).isToasted().thenReturn(False)
        when(self.mockedTortilla).getCurrentTemperature().thenReturn(
            2, 8, 8, 8, 14)
        when(self.mockedTortilla).getToastTemperature().thenReturn(20)
        when(self.mockedQueso).getCurrentTemperature().thenReturn(
            2, 8, 8, 8, 14)
        when(self.mockedQueso).getMeltingTemperature().thenReturn(10)
        self.assertEquals("Good quesadilla",
                          self.quesadilla.prepareSingle())
        verify(self.mockedTortilla, times=0).toast(True)
        verify(self.mockedQueso, times=1).melt(True)

    def test_quesadillaTerrible(self):
        when(self.mockedQueso).isMelted().thenReturn(False)
        when(self.mockedTortilla).isToasted().thenReturn(True)
        when(self.mockedTortilla).getCurrentTemperature().thenReturn(
            2, 8, 8, 8, 14)
        when(self.mockedTortilla).getToastTemperature().thenReturn(20)
        when(self.mockedQueso).getCurrentTemperature().thenReturn(
            2, 8, 8, 8, 14)
        when(self.mockedQueso).getMeltingTemperature().thenReturn(10)
        self.assertEquals("Terrible quesadilla",
                          self.quesadilla.prepareSingle())
        verify(self.mockedTortilla, times=0).toast(True)
        verify(self.mockedQueso, times=1).melt(True)

    def test_noHayGas(self):
        when(self.mockedQueso).isMelted().thenReturn(False)
        when(self.mockedTortilla).isToasted().thenReturn(False)
        when(self.mockedTortilla).getCurrentTemperature().thenReturn(
            2, 8, 8, 8, 14)
        when(self.mockedTortilla).getToastTemperature().thenReturn(20)
        when(self.mockedQueso).getCurrentTemperature().thenReturn(
            2, 8, 8, 8, 14)
        when(self.mockedQueso).getMeltingTemperature().thenReturn(10)
        self.assertEquals("You ran out of gas",
                          self.quesadilla.prepareSingle())
        verify(self.mockedTortilla, times=0).toast(True)
        verify(self.mockedQueso, times=1).melt(True)

    def test_quesadillaDoblePerfecta(self):
        when(self.mockedQuesoDoble).isMelted().thenReturn(True)
        when(self.mockedTortilla1).isToasted().thenReturn(True)
        when(self.mockedTortilla2).isToasted().thenReturn(True)
        when(self.mockedTortilla1).getCurrentTemperature().thenReturn(
            2, 8, 8, 8, 14)
        when(self.mockedTortilla1).getToastTemperature().thenReturn(10)
        when(self.mockedTortilla2).getCurrentTemperature().thenReturn(
            2, 8, 8, 8, 14)
        when(self.mockedTortilla2).getToastTemperature().thenReturn(10)
        when(self.mockedQuesoDoble).getCurrentTemperature().thenReturn(
            2, 8, 8, 8, 14)
        when(self.mockedQuesoDoble).getMeltingTemperature().thenReturn(10)
        self.assertEquals("Perfect double quesadilla",
                          self.quesadillaDoble.prepareDouble())
        verify(self.mockedTortilla1, times=1).toast(True)
        verify(self.mockedTortilla2, times=1).toast(True)
        verify(self.mockedQuesoDoble, times=1).melt(True)

    def test_quesadillaDobleBuenaTortilla1(self):
        when(self.mockedQuesoDoble).isMelted().thenReturn(True)
        when(self.mockedTortilla1).isToasted().thenReturn(False)
        when(self.mockedTortilla2).isToasted().thenReturn(True)
        when(self.mockedTortilla1).getCurrentTemperature().thenReturn(
            2, 8, 8, 8, 14)
        when(self.mockedTortilla1).getToastTemperature().thenReturn(20)
        when(self.mockedTortilla2).getCurrentTemperature().thenReturn(
            2, 8, 8, 8, 14)
        when(self.mockedTortilla2).getToastTemperature().thenReturn(10)
        when(self.mockedQuesoDoble).getCurrentTemperature().thenReturn(
            2, 8, 8, 8, 14)
        when(self.mockedQuesoDoble).getMeltingTemperature().thenReturn(10)
        self.assertEquals("Good double quesadilla toasted2",
                          self.quesadillaDoble.prepareDouble())
        verify(self.mockedTortilla1, times=0).toast(True)
        verify(self.mockedTortilla2, times=1).toast(True)
        verify(self.mockedQuesoDoble, times=1).melt(True)

    def test_quesadillaDobleBuenaTortilla2(self):
        when(self.mockedQuesoDoble).isMelted().thenReturn(True)
        when(self.mockedTortilla1).isToasted().thenReturn(True)
        when(self.mockedTortilla2).isToasted().thenReturn(False)
        when(self.mockedTortilla1).getCurrentTemperature().thenReturn(
            2, 8, 8, 8, 14)
        when(self.mockedTortilla1).getToastTemperature().thenReturn(10)
        when(self.mockedTortilla2).getCurrentTemperature().thenReturn(
            2, 8, 8, 8, 14)
        when(self.mockedTortilla2).getToastTemperature().thenReturn(20)
        when(self.mockedQuesoDoble).getCurrentTemperature().thenReturn(
            2, 8, 8, 8, 14)
        when(self.mockedQuesoDoble).getMeltingTemperature().thenReturn(10)
        self.assertEquals("Good double quesadilla toasted1",
                          self.quesadillaDoble.prepareDouble())
        verify(self.mockedTortilla1, times=1).toast(True)
        verify(self.mockedTortilla2, times=0).toast(True)
        verify(self.mockedQuesoDoble, times=1).melt(True)

    def test_quesadillaDobleRegular(self):
        when(self.mockedQuesoDoble).isMelted().thenReturn(True)
        when(self.mockedTortilla1).isToasted().thenReturn(False)
        when(self.mockedTortilla2).isToasted().thenReturn(False)
        when(self.mockedTortilla1).getCurrentTemperature().thenReturn(
            2, 8, 8, 8, 14)
        when(self.mockedTortilla1).getToastTemperature().thenReturn(20)
        when(self.mockedTortilla2).getCurrentTemperature().thenReturn(
            2, 8, 8, 8, 14)
        when(self.mockedTortilla2).getToastTemperature().thenReturn(20)
        when(self.mockedQuesoDoble).getCurrentTemperature().thenReturn(
            2, 8, 8, 8, 14)
        when(self.mockedQuesoDoble).getMeltingTemperature().thenReturn(10)
        self.assertEquals("Regular double quesadilla",
                          self.quesadillaDoble.prepareDouble())
        verify(self.mockedTortilla1, times=0).toast(True)
        verify(self.mockedTortilla2, times=0).toast(True)
        verify(self.mockedQuesoDoble, times=1).melt(True)

    def test_quesadillaDobleMala(self):
        when(self.mockedQuesoDoble).isMelted().thenReturn(False)
        when(self.mockedTortilla1).isToasted().thenReturn(True)
        when(self.mockedTortilla2).isToasted().thenReturn(True)
        when(self.mockedTortilla1).getCurrentTemperature().thenReturn(
            2, 8, 8, 8, 14)
        when(self.mockedTortilla1).getToastTemperature().thenReturn(10)
        when(self.mockedTortilla2).getCurrentTemperature().thenReturn(
            2, 8, 8, 8, 14)
        when(self.mockedTortilla2).getToastTemperature().thenReturn(10)
        when(self.mockedQuesoDoble).getCurrentTemperature().thenReturn(
            2, 8, 8, 8, 14)
        when(self.mockedQuesoDoble).getMeltingTemperature().thenReturn(10)
        self.assertEquals("Bad double quesadilla",
                          self.quesadillaDoble.prepareDouble())
        verify(self.mockedTortilla1, times=1).toast(True)
        verify(self.mockedTortilla2, times=1).toast(True)
        verify(self.mockedQuesoDoble, times=1).melt(True)

    def test_quesadillaDobleTerrible1(self):
        when(self.mockedQuesoDoble).isMelted().thenReturn(False)
        when(self.mockedTortilla1).isToasted().thenReturn(False)
        when(self.mockedTortilla2).isToasted().thenReturn(True)
        when(self.mockedTortilla1).getCurrentTemperature().thenReturn(
            2, 8, 8, 8, 14)
        when(self.mockedTortilla1).getToastTemperature().thenReturn(20)
        when(self.mockedTortilla2).getCurrentTemperature().thenReturn(
            2, 8, 8, 8, 14)
        when(self.mockedTortilla2).getToastTemperature().thenReturn(10)
        when(self.mockedQuesoDoble).getCurrentTemperature().thenReturn(
            2, 8, 8, 8, 14)
        when(self.mockedQuesoDoble).getMeltingTemperature().thenReturn(10)
        self.assertEquals("Terrible double quesadilla toasted2",
                          self.quesadillaDoble.prepareDouble())
        verify(self.mockedTortilla1, times=0).toast(True)
        verify(self.mockedTortilla2, times=1).toast(True)
        verify(self.mockedQuesoDoble, times=1).melt(True)

    def test_quesadillaDobleTerrible2(self):
        when(self.mockedQuesoDoble).isMelted().thenReturn(False)
        when(self.mockedTortilla1).isToasted().thenReturn(True)
        when(self.mockedTortilla2).isToasted().thenReturn(False)
        when(self.mockedTortilla1).getCurrentTemperature().thenReturn(
            2, 8, 8, 8, 14)
        when(self.mockedTortilla1).getToastTemperature().thenReturn(10)
        when(self.mockedTortilla2).getCurrentTemperature().thenReturn(
            2, 8, 8, 8, 14)
        when(self.mockedTortilla2).getToastTemperature().thenReturn(20)
        when(self.mockedQuesoDoble).getCurrentTemperature().thenReturn(
            2, 8, 8, 8, 14)
        when(self.mockedQuesoDoble).getMeltingTemperature().thenReturn(10)
        self.assertEquals("Terrible double quesadilla toasted1",
                          self.quesadillaDoble.prepareDouble())
        verify(self.mockedTortilla1, times=1).toast(True)
        verify(self.mockedTortilla2, times=0).toast(True)
        verify(self.mockedQuesoDoble, times=1).melt(True)

    def test_quesadillaDobleNoSeCocino(self):
        when(self.mockedQuesoDoble).isMelted().thenReturn(False)
        when(self.mockedTortilla1).isToasted().thenReturn(False)
        when(self.mockedTortilla2).isToasted().thenReturn(False)
        when(self.mockedTortilla1).getCurrentTemperature().thenReturn(
            2, 8, 8, 8, 14)
        when(self.mockedTortilla1).getToastTemperature().thenReturn(20)
        when(self.mockedTortilla2).getCurrentTemperature().thenReturn(
            2, 8, 8, 8, 14)
        when(self.mockedTortilla2).getToastTemperature().thenReturn(20)
        when(self.mockedQuesoDoble).getCurrentTemperature().thenReturn(
            2, 8, 8, 8, 14)
        when(self.mockedQuesoDoble).getMeltingTemperature().thenReturn(10)
        self.assertEquals("Not cooked double quesadilla",
                          self.quesadillaDoble.prepareDouble())
        verify(self.mockedTortilla1, times=0).toast(True)
        verify(self.mockedTortilla2, times=0).toast(True)
        verify(self.mockedQuesoDoble, times=1).melt(True)
